x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/TransformInsufficientPermissionsIT.java

```diff
@@ -0,0 +1,247 @@
+            ResponseException.class,
+            () -> startTransform(config.getId(), RequestOptions.DEFAULT.toBuilder().addHeader(AUTH_KEY, JUNIOR_HEADER).build())
+        );
+        assertThat(e.getResponse().getStatusLine().getStatusCode(), is(equalTo(500)));
+        assertThat(
+            e.getMessage(),
+            containsString(
+                String.format(Locale.ROOT, "Could not create destination index [%s] for transform [%s]", destIndexName, transformId)
+            )
+        );
+
+        assertThat(extractValue(getTransformStats(transformId), "health", "status"), is(equalTo("green")));
+
+        e = expectThrows(
+            ResponseException.class,
+            () -> startTransform(config.getId(), RequestOptions.DEFAULT.toBuilder().addHeader(AUTH_KEY, SENIOR_HEADER).build())
+        );
+        assertThat(e.getResponse().getStatusLine().getStatusCode(), is(equalTo(500)));
+        assertThat(
+            e.getMessage(),
+            containsString(
+                String.format(Locale.ROOT, "Could not create destination index [%s] for transform [%s]", destIndexName, transformId)
+            )
+        );
+
+        assertThat(extractValue(getTransformStats(transformId), "health", "status"), is(equalTo("green")));
+
+        // update transform's credentials so that the transform has permission to access source/dest indices
+        updateConfig(transformId, "{}", RequestOptions.DEFAULT.toBuilder().addHeader(AUTH_KEY, SENIOR_HEADER).build());
+
+        // _start API now works
+        startTransform(config.getId(), RequestOptions.DEFAULT);
+        waitUntilCheckpoint(transformId, 1);
+
+        assertThat(extractValue(getTransformStats(transformId), "health", "status"), is(equalTo("green")));
+    }
+
+    /**
+     * defer_validation = true
+     * unattended       = true
+     */
+    @SuppressWarnings("unchecked")
+    public void testTransformPermissionsDeferValidationUnattended() throws Exception {
+        String transformId = "transform-permissions-defer-unattended";
+        String sourceIndexName = transformId + "-index";
+        String destIndexName = sourceIndexName + "-dest";
+        createReviewsIndex(sourceIndexName, 10, NUM_USERS, TransformIT::getUserIdForRow, TransformIT::getDateStringForRow);
+
+        TransformConfig config = createConfig(transformId, sourceIndexName, destIndexName, true);
+        putTransform(
+            transformId,
+            Strings.toString(config),
+            RequestOptions.DEFAULT.toBuilder().addParameter("defer_validation", String.valueOf(true)).build()
+        );
+        assertThat(extractValue(getTransformStats(transformId), "health", "status"), is(equalTo("green")));
+
+        startTransform(config.getId(), RequestOptions.DEFAULT);
+
+        Thread.sleep(5_000);
+
+        // transform is yellow
+        Map<String, Object> stats = getTransformStats(transformId);
+        assertThat(extractValue(stats, "health", "status"), is(equalTo("yellow")));
```

Comment:
It would be best to avoid a 5 second wait.

Would an `assertBusy` that the transform is `yellow` be good enough?

Or if not, is there something else we could wait for that would avoid the hardcoded delay?